### PR TITLE
docs: improve readme tests section

### PR DESCRIPTION
**Pull request by Benehiko**
## Proposed changes

Update the README.md file with the correct test commands as the previous ones were incomplete or out dated.

## Checklist

- [X] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [X] I have read the [security policy](../security/policy).
- [X] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -435,18 +435,47 @@ There are three types of tests you can run:
...
+
+    # quick tests
+    make quicktest
+
+    # regular tests
+    make test
+    test-resetdb
+
+    # end-to-end tests
+    make e2e
```
Comment (zepatrik) [inline README.md:450]: We usually use the fenced (backticks) codeblocks instead.

File: README.md [hunk 2]
```diff
@@ -435,18 +435,47 @@ There are three types of tests you can run:
...
+
 ##### Short Tests
 
+It is recommended to use the make file to run your tests using `make quicktest`
+, however, you can still use the `go test` command.
+
+**Please note**:
+
+Running quick tests will use the sqlite database
+thus add `-tags sqlite` when building the binary.
```
Comment (zepatrik) [inline README.md:461]: Even the full tests run against sqlite and require the tag.
```suggestion
All tests run against an sqlite in-memory database,
thus it is required to use the `-tags sqlite` build tag.
```

File: README.md [hunk 3]
```diff
@@ -435,18 +435,47 @@ There are three types of tests you can run:
...
 ```shell script
-go test -short ./...
+go test -v -failfast -short -tags sqlite ./...
 ```
 
 or test just a specific module:
 
 ```shell script
-cd client; go test -short .
+cd client; go test -v -failfast -short -tags sqlite .
```
Comment (zepatrik) [inline README.md:472]:
```suggestion
go test -v -failfast -short -tags sqlite ./client
```

**Review by zepatrik — CHANGES_REQUESTED** [review]
Nice, the section was quite outdated it seems :sweat_smile:

**Review by aeneasr — APPROVED** [review]

**Pull request merged**